### PR TITLE
[FLINK-10046] [Startup Shell Scripts] .bat script to start a Flink service as a console application

### PR DESCRIPTION
**Pull request by pavel-shvetsov-git**
Created .bat script to start a Flink service as a console application
This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-dist/src/main/flink-bin/bin/flink-console.bat [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+SET FLINK_CLASSPATH=%FLINK_LIB_DIR%\*
+SET FLINK_CONF_DIR=%FLINK_ROOT_DIR%\conf
+SET FLINK_LOG_DIR=%FLINK_ROOT_DIR%\log
+SET JVM_ARGS=-Xms1024m -Xmx1024m
+
+SET logname_tm=flink-%username%-taskmanager.log
+SET log_tm=%FLINK_LOG_DIR%\%logname_tm%
+SET outname_tm=flink-%username%-taskmanager.out
+SET out_tm=%FLINK_LOG_DIR%\%outname_tm%
+SET log_setting_tm=-Dlog.file="%log_tm%" -Dlogback.configurationFile=file:"%FLINK_CONF_DIR%\logback.xml" -Dlog4j.configuration=file:"%FLINK_CONF_DIR%\log4j.properties"
```
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/flink-console.bat]: remove `_tm` suffixes

File: flink-dist/src/main/flink-bin/bin/flink-console.bat [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+:: Log rotation (quick and dirty)
+CD "%FLINK_LOG_DIR%"
+for /l %%x in (5, -1, 1) do (
+SET /A y = %%x+1
+RENAME "%logname_jm%.%%x" "%logname_jm%.!y!" 2> nul
```
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/flink-console.bat]: `logname_jm` is undefined

File: flink-dist/src/main/flink-bin/bin/flink-console.bat [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+RENAME "%logname_jm%" "%logname_jm%.0"  2> nul
+RENAME "%logname_tm%" "%logname_tm%.0"  2> nul
+RENAME "%outname_jm%" "%outname_jm%.0"  2> nul
+RENAME "%outname_tm%" "%outname_tm%.0"  2> nul
+DEL "%logname_jm%.6"  2> nul
+DEL "%logname_tm%.6"  2> nul
+DEL "%outname_jm%.6"  2> nul
+DEL "%outname_tm%.6"  2> nul
+
+IF "%SERVICE%"=="jobmanager" (
```
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/flink-console.bat]: input validation should always be done first to fail early.

File: flink-dist/src/main/flink-bin/bin/flink-console.bat [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+) ELSE IF "%SERVICE%"=="taskexecutor" (
+    SET CLASS_TO_RUN=org.apache.flink.runtime.taskexecutor.TaskManagerRunner
+) ELSE IF "%SERVICE%"=="historyserver" (
+    SET CLASS_TO_RUN=org.apache.flink.runtime.webmonitor.history.HistoryServer
+) ELSE IF "%SERVICE%"=="zookeeper" (
+    SET CLASS_TO_RUN=org.apache.flink.runtime.zookeeper.FlinkZooKeeperQuorumPeer
+) ELSE IF "%SERVICE%"=="standalonesession" (
+    SET CLASS_TO_RUN=org.apache.flink.runtime.entrypoint.StandaloneSessionClusterEntrypoint
+) ELSE (
+    ECHO Unknown service %SERVICE%. Usage: flink-console.bat ^(jobmanager^|taskmanager^|historyserver^|zookeeper^) ^[args^]
```
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/flink-console.bat]: doesn't list all possible services

**Comment by aljoscha** [comment]
The Flink developers have decided that we don't want to extend windows script support for now: https://lists.apache.org/thread.html/r7693d0c06ac5ced9a34597c662bcf37b34ef8e799c32cc0edee373b2%40%3Cdev.flink.apache.org%3E. I'm very sorry because you already invested time in this.

**Pull request closed without merging**